nixos/doc/manual/release-notes/rl-2405.section.md

```diff
@@ -399,6 +399,10 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 
 - `halloy` package was updated past 2024.5 which introduced a breaking change by switching the config format from YAML to TOML. See https://github.com/squidowl/halloy/releases/tag/2024.5 for details.
 
+- If `services.smokeping.webService` was enabled, smokeping is now served via nginx instead of thttpd. This change brings the following consequences:
+  - The default port for smokeping is now the nginx default port 80 instead of 8081.
+  - The option `services.smokeping.port` has been removed. To customize the port, use `services.nginx.virtualHosts.smokeping.listen.*.port`.
+
```

Comment:
Why couldn't services.smokeping.port be kept (option and value)? It seems the interface was abstract, and thttpd could be swapped for nginx without any user visible changes, but now users have to know the implementation detail of the nginx backend?

Reply:
That essentially made it impossible to use the SSL functionalities the nginx vhosts did provide, or at least very unpredictable. We would need to carry a lot of complicated fallback logic, just for the sake of still providing the old port option. The maintenance overhead stood in no relation to the usefulness of that port option.

Moving this to plain nginx vhosts makes things much more predictable, using a well-maintained stack. We don't need another webserver, it's now just nginx vhost config, and if you want to use something else, you still can, configuring something similar to the nginx config in your webserver.

Reply:
To be more specific, nginx offers an option to enable ssl for all your vhosts at once, which would break by having this separate port option.